zerver/views/users.py

```diff
@@ -91,17 +92,32 @@ def update_user_backend(request: HttpRequest, user_profile: UserProfile, user_id
     # This condition protected against situations where update_user_backend
     # could cause both is_guest and is_realm_admin to be set.
     # Once we update the frontend to just send a 'role' value, we can remove this check.
-    if (((is_guest is None and target.is_guest) or is_guest) and
-            ((is_admin is None and target.is_realm_admin) or is_admin)):
-        return json_error(_("Guests cannot be organization administrators"))
+    if ((is_guest is None and target.is_guest) or is_guest):
+        if ((is_admin is None and target.is_realm_admin and not target.is_realm_owner) or is_admin):
+            return json_error(_("Guests cannot be organization administrators"))
+
+        if ((is_owner is None and target.is_realm_owner) or is_owner):
+            return json_error(_("Guests cannot be organization owners"))
+
+    if (((is_owner is None and target.is_realm_owner) or is_owner) and
+            ((is_admin is None and target.is_realm_admin and not target.is_realm_owner) or is_admin)):
+        return json_error(_("Organization administrators cannot be organization owners"))
```

Comment:
This checks ensures two things-
- More than one value cannot be true.
- Existing role should also be set to false along with the new role passed as true.